src/config/mod.rs

```diff
@@ -54,12 +55,25 @@ pub struct Config {
 
 pub fn load_configs(filenames: &Vec<PathBuf>) -> Result<Config, Box<dyn error::Error>> {
     // Assumes filenames is non-empty
-    let yaml = fs::read_to_string(&filenames[0])?;
-    let mut config: Config = serde_yaml::from_str(&yaml)?;
+    let config_contents = fs::read_to_string(&filenames[0])?;
+    let mut is_yaml = true;
+
+    let mut config: Config = {
+        match serde_yaml::from_str(&config_contents) {
+            Ok(c) => c,
+            Err(_) => { is_yaml = false; toml::from_str(&config_contents)? }
```

Comment:
When you use a YAML file and there's a syntax error in it, this would report a syntax error as a TOML file. Can we just look at the filename's extension and use TOML only when it ends with `.toml` and use YAML if otherwise (because it'd be backward-compatible when there's no extension and things like `.json` is also a valid YAML file)?